rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -235,14 +235,38 @@ Node::set_parameters_atomically(
 {
   std::lock_guard<std::mutex> lock(mutex_);
   std::map<std::string, rclcpp::parameter::ParameterVariant> tmp_map;
+  std::vector<rcl_interfaces::msg::Parameter> new_parameters;
+  std::vector<rcl_interfaces::msg::Parameter> changed_parameters;
+  std::vector<rcl_interfaces::msg::ParameterDescriptor> deleted_parameters;
+
   for (auto p : parameters) {
+    if (parameters_.find(p.get_name()) == parameters_.end()) {
+      new_parameters.push_back(p.to_parameter());
```

Comment:
Shouldn't this block check for the parameter type? And if it is `not-set` not add the parameter?